src/libraries/System.Runtime.Extensions/tests/System/Random.cs

```diff
@@ -753,9 +753,52 @@ public static void Shuffle_Span_Seeded(bool emptyShuffle)
             }
 
             random.Shuffle(items);
-            AssertExtensions.SequenceEqual(stackalloc int[] { 1, 4, 3, 2 }, items);
+            AssertExtensions.SequenceEqual(stackalloc int[] { 2, 1, 4, 3 }, items);
         }
 
+        [Theory]
+        [InlineData(1000, 50, 2565)]
+        [InlineData(1000, 49, 2465)]
+        public static void Shuffle_Array_Fairness(int runs, int len, double critical)
+        {
+          // Test that repeatedly shuffling an array puts each value in each position a roughly equal number of times.
+          // This does not prove that the shuffle is fair but will at least fail if the shuffle is obviously unfair.
+          // This is a "Chi Squared Goodness of fit" problem.
+          // The number of degrees of freedom is (len - 1)*(len - 1).
+          // The significance is 0.01 so this test should naturally fail for 1 in 100 seeds.
```

Comment:
that sounds like a flaky test? These tests run 1000's of times a day someplace or another. I was using a critical value to give 10**-8 chance of failure. That was still enough to be basically certain of failure if there was a serious bug.

Reply:
I've used a fixed seed so this test will always produce the same result no matter how many times you run it. It might falsely fail if the rng or the shuffle implementation changes but if that happens, one can change the seed and 99% of seeds should work.